File: Classes/Objects/catalog.vb
''' <summary>
''' Description of form/class/etc.
''' </summary>
Public Class catalog
	Private itemDatabase() As item
	Private itemCurrent As item
	Public Sub New()
		
	End Sub
	
	'Conains an array all items.
	Public Sub LoadDatabase()
		Dim stringItemPaths() As String
		stringItemPaths = System.IO.Directory.GetFiles(System.IO.Directory.GetCurrentDirectory & "\data\items\store", "*.ini",  System.IO.SearchOption.AllDirectories)
		ReDim itemDatabase(stringItemPaths.GetUpperBound(0))
		
		Dim integerEntry As Integer = 0
		For Each itempath As String In stringItemPaths
			itemDatabase(integerEntry) = New item(itempath)
			integerEntry += 1
		Next
	End Sub
	
	'Get return with Family,Category,Genre,Name
	Public Function GetItem(Family As String, Category As String, Genre As String, Name As String) As item
		For Each article As item In itemDatabase
			If article.Family = Family AndAlso article.Category = Category AndAlso article.Genre = Genre AndAlso article.Name_Title = Name Then
				'Get article to itemCurrent & Return.
				itemCurrent = article
				Return article
				Exit Function
			End If
		Next
		MsgBox("No such item in database." & Chr(13) & Family & " - " & Category & " - " & Genre & " - " & Name)
		Return Nothing
	End Function
End Class

File: Game/Classes/Objects/catalog.cs
using Microsoft.VisualBasic;
using System;

/// <summary>
/// Description of form/class/etc.
/// </summary>
public class catalog
{
    private item[] itemDatabase;
    private item itemCurrent;
    public catalog()
    {

    }

    //Conains an array all items.
    public void LoadDatabase()
    {
        string[] stringItemPaths = null;
        stringItemPaths = System.IO.Directory.GetFiles(System.IO.Directory.GetCurrentDirectory + "\\data\\items\\store", "*.ini", System.IO.SearchOption.AllDirectories);
        itemDatabase = new item[stringItemPaths.GetUpperBound(0) + 1];

        int integerEntry = 0;
        foreach (string itempath in stringItemPaths) {
            itemDatabase[integerEntry] = new item(itempath);
            integerEntry += 1;
        }
    }

    //Get return with Family,Category,Genre,Name
    public item GetItem(string Family, string Category, string Genre, string Name)
    {
        foreach (item article in itemDatabase) {
            if (article.Family == Family && article.Category == Category && article.Genre == Genre && article.Name_Title == Name) {
                //Get article to itemCurrent & Return.
                itemCurrent = article;
                return article;
            }
        }
        Interaction.MsgBox("No such item in database." + Char(13) + Family + " - " + Category + " - " + Genre + " - " + Name);
        return null;
    }
}